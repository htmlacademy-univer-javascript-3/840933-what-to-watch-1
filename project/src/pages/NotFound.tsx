import { Link } from 'react-router-dom';

export const NotFound = () => (
  <section className="not-found-page__container">
    <h1>404</h1>
    <p>Страница не найдена</p>
    <Link to="/">Вернуться на главную</Link>
  </section>
);
