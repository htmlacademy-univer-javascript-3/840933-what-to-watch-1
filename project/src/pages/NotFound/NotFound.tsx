import { Logo } from '../../components/Logo/Logo';

export const NotFound = () => (
  <div className="user-page">
    <header className="page-header user-page__head">
      <Logo light={false} />
      <h1 className="page-title user-page__title">Not Found</h1>
    </header>
  </div>
);
