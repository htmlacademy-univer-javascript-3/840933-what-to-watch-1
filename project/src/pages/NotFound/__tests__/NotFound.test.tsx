import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';

import { NotFound } from '../NotFound';

const initialEntries = ['/'];

describe('<NotFound />', () => {
  it('should render', () => {
    const view = render(
      <MemoryRouter initialEntries={initialEntries}>
        <NotFound />
      </MemoryRouter>
    );

    expect(view).toMatchSnapshot();

    expect(
      screen.getByText('Not Found')
    ).toBeInTheDocument();
  });
});
