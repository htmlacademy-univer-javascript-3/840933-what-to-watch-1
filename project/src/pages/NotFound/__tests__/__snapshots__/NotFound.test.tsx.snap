// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="user-page"
      >
        <header
          class="page-header user-page__head"
        >
          <div
            class="logo"
          >
            <a
              class="logo__link "
              href="/"
            >
              <span
                class="logo__letter logo__letter--1"
              >
                W
              </span>
              <span
                class="logo__letter logo__letter--2"
              >
                T
              </span>
              <span
                class="logo__letter logo__letter--3"
              >
                W
              </span>
            </a>
          </div>
          <h1
            class="page-title user-page__title"
          >
            Not Found
          </h1>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="user-page"
    >
      <header
        class="page-header user-page__head"
      >
        <div
          class="logo"
        >
          <a
            class="logo__link "
            href="/"
          >
            <span
              class="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              class="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              class="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <h1
          class="page-title user-page__title"
        >
          Not Found
        </h1>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
